lib/cerner/oauth1a/oauth_error.rb

```diff
@@ -27,19 +30,23 @@ class OAuthError < StandardError
       # oauth_parameters   - A String/Symbol or Array of Strings/Symbols containing the names of parameters that
       #                      are absent or rejected. This is should only be used when oauth_problem
       #                      is 'parameter_absent' or 'parameter_rejected' Optional.
+      # realm              - The protection realm associated with the error. Optional.
       def initialize(
```

Comment:
This initialize method is getting a bit unwieldy with the new realm parameter, as can be seen in all the places this is called with a realm value, but nil for many of the other parameters.

It seems like using named parameters would be ideal here, but I believe that would be a breaking change. Is there a good alternative for constructing an error with different subsets of parameters?

Reply:
No, there's not a great way to passively change initialize methods. At least, I've not found a great way. I'm resigned to this at the moment since there's no reason for any code to use this directly, they should only be catching the error.